File: server/guess/src/main/java/com/alloymobile/guess/persistence/dbo/Game.java
package com.alloymobile.guess.persistence.dbo;

import lombok.Data;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Data
@ToString(exclude = {"roundGame","wordGame","teamGame"})
public class Game implements IGuessDBO {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String score;

    @ManyToOne(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST,CascadeType.MERGE})
    @JoinColumn(name="ROUND_ID")
    private Round roundGame;

    @OneToOne(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST,CascadeType.MERGE})
    @JoinColumn(name="WORD_ID")
    private Word wordGame;

    @ManyToOne(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST,CascadeType.MERGE})
    @JoinColumn(name="TEAM_ID")
    private Team teamGame;
}

File: client/guess/src/app/model/game.ts
import { Word } from './word';
import { Team } from './team';
import { Round } from './round';

export class Game {
    id: number;
    word: Word;
    team: Team;
    round: Round;
    constructor(game?: Game){
        if (game) {
            this.id = game.id;
            this.word = new Word(game.word);
            this.team = new Team(game.team);
            this.round = new Round(game.round);
        } else {
            this.id = 0;
            this.word = new Word();
            this.team = new Team();
            this.round = new Round();
        }
    }
}
